In [1]:
import pandas as pd
pd.options.display.max_rows = 150

## Exploring the data

### USDA data set
[Unemployment and median household income for the United States, States, and counties, 2000–23](https://www.ers.usda.gov/data-products/county-level-data-sets/county-level-data-sets-download-data)

In [2]:
df_USDA = pd.read_csv('Data/CountyIncomeUSDA/Unemployment2023.csv')

In [3]:
df_USDA.info()

<class 'pandas.DataFrame'>
RangeIndex: 329726 entries, 0 to 329725
Data columns (total 5 columns):
 #   Column     Non-Null Count   Dtype  
---  ------     --------------   -----  
 0   FIPS_Code  329726 non-null  int64  
 1   State      329726 non-null  str    
 2   Area_Name  329726 non-null  str    
 3   Attribute  329726 non-null  str    
 4   Value      329726 non-null  float64
dtypes: float64(1), int64(1), str(3)
memory usage: 12.6 MB


Federal Information Processing System (FIPS) codes identify geographic areas by state and county.
[FIPS Code details](https://transition.fcc.gov/oet/info/maps/census/fips/fips.txt)

In [4]:
df_USDA.head(10)

,FIPS_Code,State,Area_Name,Attribute,Value
0,0,US,United States,Civilian_labor_force_2000,142601576.0
1,0,US,United States,Employed_2000,136904853.0
2,0,US,United States,Unemployed_2000,5696723.0
3,0,US,United States,Unemployment_rate_2000,4.0
4,0,US,United States,Civilian_labor_force_2001,143786537.0
5,0,US,United States,Employed_2001,136977996.0
6,0,US,United States,Unemployed_2001,6808541.0
7,0,US,United States,Unemployment_rate_2001,4.7
8,0,US,United States,Civilian_labor_force_2002,144839298.0
9,0,US,United States,Employed_2002,136455783.0


In [7]:
# Pull attributes so I can explore 
df_attributes = df_USDA['Attribute']
df_attributes.value_counts()

Attribute
Civilian_labor_force_2021                    3273
Employed_2021                                3273
Unemployed_2021                              3273
Unemployment_rate_2021                       3273
Civilian_labor_force_2022                    3273
Employed_2022                                3273
Unemployed_2022                              3273
Unemployment_rate_2022                       3273
Civilian_labor_force_2023                    3273
Employed_2023                                3273
Unemployed_2023                              3273
Unemployment_rate_2023                       3273
Civilian_labor_force_2010                    3272
Employed_2010                                3272
Unemployed_2010                              3272
Unemployment_rate_2010                       3272
Civilian_labor_force_2011                    3272
Employed_2011                                3272
Unemployed_2011                              3272
Unemployment_rate_2011                  

In [6]:
# Remove year so I can see genneral types of info available
df_attributes = df_attributes.replace(r'_[0-9]+', '', regex=True).str.strip()
df_attributes.value_counts()

Attribute
Civilian_labor_force                    78419
Employed                                78419
Unemployed                              78419
Unemployment_rate                       78419
Rural_Urban_Continuum_Code               3221
Metro                                    3221
Urban_Influence_Code                     3219
Median_Household_Income                  3195
Med_HH_Income_Percent_of_State_Total     3194
Name: count, dtype: int64

Attributes available by various years
Civilian_labor_force                    78419
Employed                                78419
Unemployed                              78419
Unemployment_rate                       78419
Rural_Urban_Continuum_Code               3221
Metro                                    3221
Urban_Influence_Code                     3219
Median_Household_Income                  3195
Med_HH_Income_Percent_of_State_Total     3194